BridgeCareApp/BridgeCareCodeFirst/Models/DetailedReport.cs

```diff
@@ -0,0 +1,96 @@
+            BridgeCareContext db = new BridgeCareContext();
+            var yearsForBudget = new List<YearlyData>();
+            try
+            {
+                yearsForBudget = db.YEARLYINVESTMENTs.AsNoTracking().Where(_ => _.SIMULATIONID == data.SimulationId)
+                                                              .OrderBy(year => year.YEAR_)
+                                                              .Select(p => new YearlyData
+                                                              {
+                                                                  Year = p.YEAR_,
+                                                                  Amount = p.AMOUNT,
+                                                                  BudgetName = p.BUDGETNAME
+                                                              }).ToList();
+            }
+            catch (SqlException ex)
+            {
+                db.Dispose();
+                if (ex.Number == -2 || ex.Number == 11)
+                {
+                    throw new TimeoutException("The server has timed out. Please try after some time");
+                }
+                if (ex.Number == 208)
+                {
+                    throw new InvalidOperationException("Years data does not exist in the database");
+                }
+            }
+            db.Dispose();
+            return yearsForBudget;
+        }
+
+        public IQueryable<DetailedReport> GetDataForReport(SimulationResult data, BridgeCareContext db)
+        {
+            IQueryable<DetailedReport> RawQueryForData = null;
+            string getReport =
+                    "SELECT Facility, Section, Treatment, NumberTreatment, IsCommitted, Years " +
+                        " FROM Report_" + data.NetworkId
+                        + "_" + data.SimulationId + " Rpt WITH (NOLOCK) INNER JOIN Section_" + data.NetworkId + " Sec WITH (NOLOCK) " +
+                        " ON Rpt.SectionID = Sec.SectionID " +
+                        "Order By Facility, Section, Years";
+
+            try
+            {
+                RawQueryForData = db.Database.SqlQuery<DetailedReport>(getReport).AsQueryable();
+            }
+            catch (SqlException ex)
+            {
+                db.Dispose();
+                if (ex.Number == -2 || ex.Number == 11)
+                {
+                    throw new TimeoutException("The server has timed out. Please try after some time");
+                }
+                if (ex.Number == 208)
+                {
+                    throw new InvalidOperationException("Network or simulation table does not exist in the database");
+                }
+            }
+            catch (OutOfMemoryException)
+            {
+                db.Dispose();
+                throw new OutOfMemoryException("The server is out of memory. Please try after some time");
+            }
+            return RawQueryForData;
+        }
+
+        public class YearlyData
```

Comment:
This should be private

Reply:
![image](https://user-images.githubusercontent.com/21979229/51935992-d7109700-23cc-11e9-8edc-8a44f03d056a.png)

I cannot make it private because it is a return type for a public method. Please check out the compile time error

Reply:
As we discussed move the YearlyData class into its own sibling class. i.e. Do not subclass it.

Reply:
Done